src/classes/ContactsInLegacyHouseholds.cls

```diff
@@ -0,0 +1,594 @@
+                            contactsThatDoNotHaveAHousehold));
+        }
+        insertHouseholdsForContacts(contactsNeedingHouseholds, false);
+    }
+
+    private Set<Id> getOldNonEmptyHouseholdIds() {
+        Set<Id> oldNonEmptyHouseholdIds = new Set<Id>();
+        for (Contact con : (List<Contact>) getRecords()) {
+            if (getOldHouseholdSizeById().get(oldHouseholdIdFor(con)) > 0) {
+                oldNonEmptyHouseholdIds.add(oldHouseholdIdFor(con));
+            }
+        }
+        return oldNonEmptyHouseholdIds;
+    }
+
+    private List<Contact> contactsThatChangedHouseholds() {
+        List<Contact> contactsThatChangedHouseholds = new List<Contact>();
+        for (Contact con : (List<Contact>) getRecords()) {
+            if (householdChanged(con)) {
+                contactsThatChangedHouseholds.add(con);
+            }
+        }
+        return contactsThatChangedHouseholds;
+    }
+
+    private Set<Id> householdIdsFor(List<Contact> contacts) {
+        Set<Id> householdIds = new Set<Id>();
+        for (Contact con : contacts) {
+            householdIds.add(householdIdFor(con));
+        }
+        return householdIds;
+    }
+
+    private fflib_SObjects contactsThatDoNotHaveAHousehold() {
+        return selectByFieldValue(Contact.npo02__Household__c, null);
+    }
+
+    private List<Contact> contactsInIndividualAccounts(List<Contact> contacts) {
+        Map<Id, Account> contactAccountMap = getContactAccountMap(contacts);
+        List<Contact> contactsInIndividualAccounts = new List<Contact>();
+        for (Contact con : contacts) {
+            if (hasAccount(con)) {
+                Account acc = contactAccountMap.get(con.AccountId);
+                if (isIndividualAccount(acc)) {
+                    contactsInIndividualAccounts.add(con);
+                }
+            } else {
+                contactsInIndividualAccounts.add(con);
+            }
+        }
+        return contactsInIndividualAccounts;
+    }
+
+    private Boolean hasAccount(Contact con) {
+        return con.AccountId != null;
+    }
+
+    private Boolean isIndividualAccount(Account acc) {
+        return acc.npe01__SYSTEMIsIndividual__c == true;
+    }
+
+    private Map<Id, Integer> oldHouseholdSizeById;
+    private Map<Id, Integer> getOldHouseholdSizeById() {
+        if (oldHouseholdSizeById != null) {
+            return oldHouseholdSizeById;
+        }
+        oldHouseholdSizeById = new Map<Id, Integer>();
+        for (AggregateResult aggregateResult : getMemberCountForOldHouseholds()) {
+            oldHouseholdSizeById.put(
+                    oldHouseholdIdFrom(aggregateResult),
+                    memberCountFrom(aggregateResult));
+        }
+        for (Id hhid : getOldHouseholdIds()) {
+            if (!oldHouseholdSizeById.containskey(hhid))
+                oldHouseholdSizeById.put(hhid, 0);
+        }
+        return oldHouseholdSizeById;
+    }
+
+    private Integer memberCountFrom(AggregateResult a) {
```

Comment:
I've tried a few times now to get this factored out into a class but I keep getting halfway and then just feeling funny about it.  But I hear what you are saying and will try again.  If I can't get this into a good abstraction today then it might be helpful to jump on a call and refactor it together!

Reply:
Fair enough. It's easy for me to read the code in review and say refactor, but reality on the ground (or in the ide) may be a bit different. I think you're out next week, but we can talk about this more when you're back - maybe even a little screen share coding - to see if there's a way to pull it off.

Reply:
I'm going to resolve this convo for now - please see the other comment about how I removed the Agg query altogether.